ibis/common/scope.py

```diff
@@ -0,0 +1,65 @@
+""" Module for scope
+    Scope is a dictionary mapping :class:`~ibis.expr.operations.Node`
+    subclass instances to concrete data, and the time context associate
+    with it(if any).
+    Scope is used in many backends to cache data for calculated ops.
+"""
+from typing import Optional
+
+from ibis.expr.typing import TimeContext
+from ibis.timecontext.util import TimeContextRelation, compare_timecontext
+
+
+def set_scope_item(op, result, timecontext: Optional[TimeContext]):
```

Comment:
This is not a "set" method. A set method should change the state of object and returns nothing.

```
def set_scope_item(scope, op, timecontext, value):
      scope[op] = {'value': result, 'timecontext': timecontext}
```

Reply:
I first name it as `scope_item` for the same reason. Now it is refactored and this is addressed.